src/sentry/eventstore/filter.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+
+
+class Filter(object):
+    conditions = []
+    filter_keys = {}
+    start = None
+    end = None
+
+    def __init__(self, start=None, end=None, conditions=None, filter_keys=None):
+        self.start = start
+        self.end = end
+
+        if conditions:
+            self.conditions = conditions
+
+        if filter_keys:
+            self.filter_keys = filter_keys
+
+    def to_snuba_args(self):
+        """
+        Convert a filter to the legacy snuba_args format
+        """
+        return {
+            'conditions': self.conditions,
+            'filter_keys': self.filter_keys,
+            'start': self.start,
+            'end': self.end,
+        }
+
+    def update_start(self, start):
```

Comment:
What's the gain in making this class mutable? 
Immutable classes tend to define the responsibility better (you cannot change the content so it means you can only use it to pass it to the snuab api when you are ready) and make them easier to debug in case of issues.

Reply:
If the Filter is immutable how would one incrementally build conditions. When building pagination or the modal summary graph we need to build conditions that reflect both the current search query and include values from the currently selected event, which will likely take two passes.